src/track/trackmetadatataglib.cpp

```diff
@@ -1010,44 +1146,45 @@ bool writeTrackMetadataIntoXiphComment(TagLib::Ogg::XiphComment* pTag,
         return false;
     }
 
-    writeTrackMetadataIntoTag(pTag, trackMetadata);
-
-    // Taglib does not support the update of Vorbis comments.
-    // thus, we have to remove the old comment and add the new one
+    writeTrackMetadataIntoTag(pTag, trackMetadata,
+            WRITE_TAG_OMIT_TRACK_NUMBER | WRITE_TAG_OMIT_YEAR);
 
-    pTag->removeField("ALBUMARTIST");
-    pTag->addField("ALBUMARTIST",
-            toTagLibString(trackMetadata.getAlbumArtist()));
+    // According to https://wiki.xiph.org/Field_names "TRACKTOTAL" is
+    // the proposed field name, but some applications use "TOTALTRACKS".
+    writeXiphCommentField(pTag, "TRACKNUMBER",
+            toTagLibString(trackMetadata.getTrackNumber()));
+    writeXiphCommentField(pTag, "TRACKTOTAL",
+            toTagLibString(trackMetadata.getTrackTotal()));
+    // Remove the secondary/alternative field for total tracks to prevent
+    // inconsistencies if the tags are edited again by another external
+    // application that is unaware of "TRACKTOTAL".
+    pTag->removeField("TOTALTRACKS");
```

Comment:
I just want to prevent a future bug "Mixxx deletes all my total track infos" which happens if an external App uses only the "TOTALTRACKS" tag. 
I would accept a future inconsistency if the user edits the total tracks in Mixxx, since it is not our responsibility that two fields exists and there is a user uses an "old" app which uses the wrong one. 

Changing the total track number is anyway a very rare action, but the current delete solution also kicks in if a user edits an unrelated field in Mixxx. 

How about update "TOTALTRACKS" if it exists or only delete "TOTALTRACKS" if the info dos not match "TRACKTOTAL".